### ISSUE TITLE
toolchain.py doesn't allow empty list in get_variable.

### ISSUE THREAD

**Issue by DirkdeDraak** [hpcugent/easybuild-framework#1431]
I was trying to build the latest Doxygen, 1.8.10. It needs CMake. ANd I want to build it with the dummy compilers.
Then it fails.

```
== temporary log file in case of crash 
== processing EasyBuild easyconfig easyconfigs/d/Doxygen/Doxygen-1.8.10.eb
== building and installing Doxygen/1.8.10...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
ERROR: Traceback (most recent call last):
  File "software/EasyBuild/2.3.0/lib/python2.6/site-packages/easybuild_framework-2.3.0-py2.6.egg/easybuild/main.py", line 112, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "software/EasyBuild/2.3.0/lib/python2.6/site-packages/easybuild_framework-2.3.0-py2.6.egg/easybuild/framework/easyblock.py", line 1994, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "software/EasyBuild/2.3.0/lib/python2.6/site-packages/easybuild_framework-2.3.0-py2.6.egg/easybuild/framework/easyblock.py", line 1934, in run_all_steps
    self.run_step(step_name, step_methods)
  File "software/EasyBuild/2.3.0/lib/python2.6/site-packages/easybuild_framework-2.3.0-py2.6.egg/easybuild/framework/easyblock.py", line 1816, in run_step
    m(self)
  File "software/EasyBuild/2.3.0/lib/python2.6/site-packages/easybuild_framework-2.3.0-py2.6.egg/easybuild/framework/easyblock.py", line 1866, in <lambda>
    configure_step_spec = (CONFIGURE_STEP, 'configuring', [lambda x: x.configure_step()], True)
  File "software/EasyBuild/2.3.0/lib/python2.6/site-packages/easybuild_easyblocks-2.3.0-py2.6.egg/easybuild/easyblocks/generic/cmakemake.py", line 64, in configure_step
    include_paths = os.pathsep.join(self.toolchain.get_variable("CPPFLAGS", list))
  File "software/EasyBuild/2.3.0/lib/python2.6/site-packages/easybuild_framework-2.3.0-py2.6.egg/easybuild/tools/toolchain/toolchain.py", line 176, in get_variable
    print self.variables[name]
KeyError: 'CPPFLAGS'
```
The error is that the CPPFLAGS are not found in the list.

The simplest thing to do is to check if the key can be found in the list:
```
--- easybuild_framework-2.3.0-py2.6.egg/easybuild/tools/toolchain/toolchain.py.orig     2015-09-23 15:25:20.608042576 +0200
+++ easybuild_framework-2.3.0-py2.6.egg/easybuild/tools/toolchain/toolchain.py  2015-10-15 14:56:26.931266670 +0200
@@ -173,6 +173,8 @@
         if typ == str:
             return str(self.variables[name])
         elif typ == list:
+            if not name in self.variables:
+               return ""
             return self.variables[name].flatten()
         else:
             raise EasyBuildError("get_variable: Don't know how to create value of type %s.", typ)
```

**Comment by boegel** [comment]
Thanks for the bug report and possible workaround @DirkdeDraak, we'll take a look at fixing this.

**Comment by boegel** [comment]
fixed in #1566, thanks for reporting and sharing your workaround @DirkdeDraak!
